src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -12,14 +11,14 @@ namespace System.Data.SqlClient.SNI
     /// </summary>
     internal class SNIMarsManager
     {
-        public static readonly SNIMarsManager Singleton = new SNIMarsManager();
-        private ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections = new ConcurrentDictionary<SNIHandle, SNIMarsConnection>();
+        private readonly ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections;
 
         /// <summary>
         /// Constructor
         /// </summary>
         public SNIMarsManager()
```

Comment:
Now that I think about it, we may not even need the MarsManager at all. There's only ever one SNIMarsConnection object per physical connection, so the extra manager logic is wasted.